assets/js/frontend/checkout.js

```diff
@@ -374,7 +374,7 @@ jQuery( function( $ ) {
 							if ( ID ) {
 								if ( $.inArray( $( this ).attr( 'type' ), [ 'checkbox', 'radio' ] ) !== -1 ) {
 									$( this ).prop( 'checked', paymentDetails[ ID ] ).change();
-								} else if ( 0 === $( this ).val().length ) {
+								} else if ( $(this).val() !== null  && 0 === $( this ).val().length ) {
```

Comment:
This change looks reasonable. The code style just needs a little changing to follow WP style guidelines. 

The conditional should be a yoda conditional and there should be spacing around `this`, e.g. `null !== $( this ).val()`. If you can please make this change we can get this merged for 3.5.0.